main.go

```diff
@@ -32,30 +32,39 @@ func main() {
 	if patterns == nil {
 		patterns = forbidigo.DefaultPatterns()
 	}
+	options := []forbidigo.Option{
+		forbidigo.OptionExcludeGodocExamples(*excludeGodocExamples),
+	}
+	linter, err := forbidigo.NewLinter(patterns, options...)
+	if err != nil {
+		log.Fatalf("Could not create linter: %s", err)
+	}
 
 	cfg := packages.Config{
 		Mode:  packages.NeedSyntax | packages.NeedName | packages.NeedFiles | packages.NeedTypes,
 		Tests: *includeTests,
 	}
+
+	// Additional information may be needed for some patterns.
+	for _, pattern := range linter.Patterns() {
```

Comment:
The linter could also just expose a `NeedsTypes()` method here so we don't have to make our parsing of the patterns public.

Reply:
The type has to be public for usage in the golangci-lint config. But I can add such a method to avoid putting this knowledge about pattern types into the main.go.